src/mongoose_credentials.erl

```diff
@@ -22,13 +22,15 @@
                            %% Each backend may require different values to be present.
                            extra :: [proplists:property()] }.
 
--spec new(jid:server()) -> mongoose_credentials:t().
-new(Server) ->
+-spec new(jid:server() | {lserver, jid:lserver()}) -> mongoose_credentials:t().
+new(Server) when is_binary(Server) ->
     case jid:nameprep(Server) of
         error -> error(nameprep_failed, [Server]);
         LServer when is_binary(LServer) ->
             #mongoose_credentials{lserver = LServer}
-    end.
+    end;
+new({lserver, LServer}) when is_binary(LServer) ->
+    #mongoose_credentials{lserver = LServer}.
```

Comment:
It is a bit unexpected that this function takes a tuple or a binary and the types are not that descriptive.

Why not just get rid of the stringprepping here?

```
new(LServer) when is_binary(LServer) ->
    #mongoose_credentials{lserver = LServer}.
```

I don't see any call to this function that actually requires stringprepping anyway. Please double-check that e.g. in the test coverage report.

Reply:
Right, that's a good point. I'll also update the spec for the function, it was taking a `jid:server()`, will made it `jid:lserver()`. This ain't enforced as they're both just binaries, but hopefully the spec should be a good directive for anyone using this function in the future.